agent/task.go

```diff
@@ -271,8 +271,21 @@ func (a *Agent) makeTaskConfig(ctx context.Context, tc *taskContext) (*model.Tas
 		}
 	}
 
+	var metadata *model.UpstreamMetadata
+	if confTask.TriggerID != "" {
+		apiMetadata, err := a.comm.GetUpstreamMetadata(ctx, tc.task)
+		if err != nil {
+			return nil, err
+		}
+		service, err := apiMetadata.ToService()
+		if err != nil {
+			return nil, err
+		}
+		metadata = service.(*model.UpstreamMetadata)
```

Comment:
at the very least, do a safety check here.

I'm super wary of calling toService methods in the agent, and think this aspect of the agent where it depends on things in the models package is a huge problem and liability, is there anyway around it?